typing/datarepr.ml

```diff
@@ -101,20 +101,13 @@ let constructor_descrs ~current_unit ty_path decl cstrs rep =
   let ty_res = newgenconstr ty_path decl.type_params in
   let variant_unboxed = Builtin_attributes.has_unboxed decl.type_attributes in
   let num_consts = ref 0 and num_nonconsts = ref 0 and num_unboxed = ref 0 in
+  let variant_data = ref None in
```

Comment:
Here we would like to have the `variant_data` shared between constructors (so that we don't recompute it for each constructor of the variant). Then we could just move the let-binding under the `List.iter` just below.

Reply:
Ah, of course... Yes, moving it below would be nice. And please include a comment on the sharing: "Gabriel out of touch" is a proxy for the people unfamiliar with the code having comprehension troubles.